### ISSUE TITLE
Removing first page causes problems with second page

### ISSUE THREAD

**Issue by mdgross** [jquery/jquery-mobile#7995]
Please test with Safari.

1. View first page
2. View second page
3. First page is removed
4. Second page content now appears behind the header. It should appear below header, like the first page. 

https://jsfiddle.net/mdgross/ukztv31w/3/

**Comment by arschmitz** [comment]
@mdgross thank you for reporting this. removing the first page is not supported this will defiantly break navigation. Closing as not a bug.

**Comment by mdgross** [comment]
@arschmitz It is not breaking the navigation. It is breaking the display of the second page.

**Comment by arschmitz** [comment]
@mdgross i was really jus commenting on what you said because your fiddle does not work because the resources are blocked from loading so i cant view your example please use are jsbin template at http://jsbin.com/ofuhaw/1/edit but either way what your doing is 100% not supported. 

```
Mixed Content: The page at 'https://fiddle.jshell.net/mdgross/ukztv31w/3/show/' was loaded over HTTPS, but requested an insecure stylesheet 'http://code.jquery.com/mobile/1.4.5/jquery.mobile-1.4.5.min.css'. This request has been blocked; the content must be served over HTTPS.
fiddle.jshell.net/:1
Mixed Content: The page at 'https://fiddle.jshell.net/mdgross/ukztv31w/3/show/' was loaded over HTTPS, but requested an insecure script 'http://code.jquery.com/jquery-1.11.1.min.js'. This request has been blocked; the content must be served over HTTPS.
fiddle.jshell.net/:1
Mixed Content: The page at 'https://fiddle.jshell.net/mdgross/ukztv31w/3/show/' was loaded over HTTPS, but requested an insecure script 'http://code.jquery.com/mobile/1.4.5/jquery.mobile-1.4.5.min.js'. This request has been blocked; the content must be served over HTTPS.
fiddle.jshell.net/:34
Uncaught ReferenceError: $ is not defined
```

**Comment by mdgross** [comment]
@arschmitz I have updated the examples with jsbin. Lots of people have been depending on this workaround for a very long time now.

https://github.com/jquery/jquery-mobile/issues/3249

I am just trying to figure out what changed in 1.4.5. It seems very strange for the header of the active page to move because of deleting a hidden page.

**Comment by mdgross** [comment]
@arschmitz Update: It seems that calling $.mobile.firstPage.remove() is removing .ui-page-header-fixed and padding-top from page B.

**Comment by mdgross** [comment]
@arschmitz Would you please take another look at this? Calling $.mobile.firstPage.remove() is removing .ui-page-header-fixed and padding-top from the active page. Doesn't it seem odd to you that removing a page would alter the css of a totally different page? I understand that $.mobile.firstPage.remove() is not supported but people depend on it for  https://github.com/jquery/jquery-mobile/issues/3249#issuecomment-21111794.

**Comment by mdgross** [comment]
@jaspermdegroot  Is there any chance you might have a look at this? I would greatly appreciate it. Everything was working perfectly in 1.4.3. Now there is a new project lead and things are broken in 1.4.4 and 1.4.5. Something very strange is happening.

**Comment by jaspermdegroot** [comment]
@mdgross 

First of all, jQuery Mobile is a team/contributors effort. Even if this was an actual regression, it has nothing to do with who is project lead.

When you inspect your test page you'll see that the padding-top is removed from the page when you remove the first page. Probably because we had to make a change between 1.4.3 and 1.4.5 to fix an issue with page padding or fixed headers. Like @arschmitz said removing the first page is currently not supported. If you want to do it anyway you have to understand that your code might break, even in a maintenance release. We don't test it.
You might be able to work around the problem by calling ```updatePagePadding()``` on the active page after you removed the first page.

**Comment by solymosi** [comment]
I just bumped into the same issue and did a little debugging. It seems like that commit 3efe6fedad0f5f259d8616f0c753a22fe5595fb6 caused the issue. The problem is with the `_destroy` function, which operates on the currently active page instead of the page that the toolbar (which is being destroyed as a consequence of calling `remove` on `$.mobile.firstPage`) is actually on.

Simply assigning `this.page` to the `page` variable instead of fetching the active page from the page container seems to solve the problem.

I'm running tests at the moment, will submit a pull request soon.

**Comment by wakiki** [comment]
@solymosi looks like your PR was merged but it won't be released until v1.5. 

Is there a fix that we can use for the time being?

For now I'm using fix from @mdgross but this doesn't work:

```
$(document).on 'pagecontainerchange.fixcache', $.mobile.pageContainer, (event, ui) ->
  if $.mobile.firstPage.is('[data-dom-cache="true"]') or $.mobile.page::options.domCache and $.mobile.firstPage.is(':not([data-dom-cache="false"])')
    $(document).off 'pagecontainerchange.fixcache'
  else if ui.prevPage != undefined
    if !($(ui.toPage).is('[data-role="dialog"]') or $(ui.prevPage).is('[data-role="dialog"]'))
      if ui.toPage.attr('data-url') != ui.prevPage.attr('data-url')
        $.mobile.firstPage.remove()
        $(document).off 'pagecontainerchange.fixcache'
  return
```

**Comment by thenewguy** [comment]
@wakiki I posted an alternate workaround that may be of interest https://github.com/jquery/jquery-mobile/issues/3249#issuecomment-183936117
